service/history/timerQueueProcessor.go

```diff
@@ -261,6 +261,11 @@ func (t *timerQueueProcessorImpl) completeTimersLoop() {
 				err := t.completeTimers()
 				if err != nil {
 					t.logger.Info("Failed to complete timers.", tag.Error(err))
+					if err == ErrShardClosed {
```

Comment:
should this be added to all callers of completeTimers, such as L256?

Reply:
I think at L256, the shard is already being shut down, meaning someone already called `t.Stop()`. Check the error and call `Stop()` again will be a no-op.

Reply:
not necessary on L256 because that code path is exiting the loop anyways (its the shutdown path).